cylc/flow/task_pool.py

```diff
@@ -298,7 +305,7 @@ def release_runahead_tasks(self):
             sequence_points = []
             for sequence in self.config.sequences:
                 point = runahead_base_point
-                for _ in range(limit):
+                for _ in range(number_limit or sequence._MAX_CACHED_POINTS):
```

Comment:
This uses a limit of 100 if `number_limit` is `None`.

Would it also be worth setting a maximum of `runahead limit = P100`?

Reply:
I don't think that we would want to enforce an upper bound on the runahead limit.

Reply:
But if a workflow has no final cycle point, this for loop hang?

Reply:
Would have to re-work the logic to permit that.

Reply:
> But if a workflow has no final cycle point, there would be an infinite number of points to loop over, wouldn't there?

The loop is over the task pool, and tasks with no prerequisites spawn sequentially (when one is released from the runahead pool, the next one is spawned into the runahead pool). So that provides some constraint, but such tasks still do proliferate pretty quickly.

Reply:
But in this line it's looping over all on-sequence points up to a certain limit, before determining the task pool. For a datetime cycling workflow with a runhead limit specified as a time interval, and no final cycle point, surely there has to be some built-in limit to the number of future points to find, otherwise the loop would hang

Reply:
Update: I've come up with a way of getting the on-sequence points up to the (integer or time interval) runahead limit without imposing the arbitrary 100 point limit - https://github.com/cylc/cylc-flow/pull/3848/commits/f6b615abf9afb74c41398369f04ab682e240b47e

Reply:
(Sorry, I didn't look at the actual code before making that comment! :grimacing: )

Reply:
Workaround looks good to me.